decentraland/unity-renderer
# fix: ECS7 NFTShape

## What does this PR change?
**changes affecting production:**
[NFTShapeHQImageHandler.cs](https://github.com/decentraland/unity-renderer/pull/3066/files#diff-d41921f7df64b5f5d604ca0b5ed29344da9ffedbc7505de25599c12d253d76e1) was modified to contain only needed references (transform and collider) instead of a `INFTShapeLoaderController` reference

**changes affecting ecs7:**
updated `NFTShape` removing everything related to colliders and visibility fields
fixed exception while updating `NFTShape` model in runtime
added renderer component for `NFTShape`
disabled `NFTShape` prefabs colliders
added/updated tests accordingly
...

## How to test the changes?

Go to: https://play.decentraland.zone/?renderer-branch=fix%2Fecs7-nftshape-colliders&position=51,89 or any other location that has NFTShapes
Check that NFT image is in low resolution while not looking at it but it switch to a higher resolution while looking at it

low resolution image while not facing image directly:
<img width="1018" alt="Screen Shot 2022-09-21 at 11 10 56" src="https://user-images.githubusercontent.com/4195708/191529775-8161d7b4-fb6b-48fd-a0ed-38c790cb07f1.png">

high resolution when pointing the camera towards the image:
<img width="1033" alt="Screen Shot 2022-09-21 at 11 11 10" src="https://user-images.githubusercontent.com/4195708/191529860-c28a1aa3-5715-4a86-8490-551984d23a53.png">

## Our Code Review Standards

https://github.com/decentraland/unity-renderer/blob/master/docs/code-review-standards.md
